### DISCUSSION TITLE
Can't receive calls ?.

### DISCUSSION THREAD

**Discussion by DOX007** [community/community#78289] [Other Feature Feedback, Questions, & Ideas]
GSI TrebleDroid, on Oukitel WP23 Pro on Unisoc T606. Rooted with latest Magisk.

I'm encountering a strange problem in meny new GSI releases and rom's. I can't receive calls ????. At first I thought the problem was in the phh treble settings, Because I have no problem making calls, send and receive sms, but the phone doesn't register incoming calls ???.

How do I solve this???.

I read on various forums that you can take files from the stock firmware, in the /system/vendor/lib64/ folder, make a Magisk module and just flash it. But that folder contains tons of files and have no idea which specific file I need to copy, ore which files is about phone calls.

libimpl-ril.so
librpmbclient.so
libwpa_client.so
If I understood correctly, these files are for sms. but who controls incoming calls?. Or is there something else I can do to be able to receive calls?.

**Comment by queenofcorgis** [comment]
If you're asking how to use the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.

**Reply by DOX007** [reply]
As you may understand, there is nothing in anyone's documentation about these problems, but have seen more frequent questions about this problem on various forums. And on github with hundreds of thousands of software creators, this should be the right place to ask this question. As a wise man once said, “_Problems are for mortal weeklings, solutions are for problem solvers_...

**Reply by thejokervinay** [reply]
Did you find a solution for it? I have the same problem.

**Reply by soncao0212-commits** [reply]
author of the article, have you found the solution, I have the same problem and I can't solve it. @Thejokervinay, I see you have the solution, can you share it with me, please

**Reply by thejokervinay** [reply]
Just do the same things you would do for the GSI but the file you need to use differs, for qcom i used a different file, for mediatek you need to use a different file, i have a device which has qcom-caf so i used ims-caf-u2.apk

**Reply by soncao0212-commits** [reply]
@thejokervinay Hi, my device is unisoc T157, what file should I use? Can I use the original ims.apk file in the original rom? I installed it but it said it conflicts with the existing package. Why is that? Where did you get the ims.apk file and how did you install it? My device has no exception, any treble rom has the same error of not receiving calls. Please help me because you are the only clue that can help me solve this problem, please.

**Reply by thejokervinay** [reply]
No all devices do not work with the same ims.apk app, that's what i told you, mediatek needs a different ims, qcom needs different one, kindly search for whichever it is that your device uses or maybe urs is unisoc maybe you can join the treble group in telegram and ask.

**Reply by soncao0212-commits** [reply]
@thejokervinay I  understand what you mean, but I want to ask how to install ims. When I install ims using the apk file, the device keeps saying it conflicts with an existing package. I hope you respond.

**Reply by thejokervinay** [reply]
For that you need to start fresh, before it even has an ims package or uninstall the ims finding it properly

**Reply by soncao0212-commits** [reply]
@thejokervinay when i started installing rom, i had not installed any ims file, i saw in treble settings that install ims not supported. then i thought of installing ims file of stock rom. but it's not simple, it reported conflict with available package, and i don't know how to remove available package, can you show me how to remove it. looking forward to your feedback

**Reply by thejokervinay** [reply]
Go to setting, then apps then find system apps in that find ims app and uninstall it

**Reply by soncao0212-commits** [reply]
@thejokervinay I tried unpacking the gsi rom, then inserting the ims files and calling related apps from the stock rom and then repacking it. However, after flashing the phone got stuck at the boot logo. What a pity, have you thought of a way to fix the rom like that?

**Reply by thejokervinay** [reply]
Not like that, install GSI then install ims apk, if it tells it conflicts you need to uninstall the imp.apk app and then install the one you have, but the way you told will need to make the GSI again and then do it, that's a longer process, this works but it stays for one reboot, but the longer process you'll need to add something to make it permanent mostly, and yes i have done it its a long process

**Reply by soncao0212-commits** [reply]
@thejokervinay although i haven't found a solution yet, thanks for the feedback, and i will let you know when i find one. thanks bro

**Reply by thejokervinay** [reply]
No, trust me that's how you need to do it. But ya i don't remember maybe i did it by building the GSI newly after implementing something.

**Comment by usmanamin3254** [comment]
The reason you can’t receive calls on your Unisoc T606 GSI isn’t about SMS or making calls—it’s specifically how incoming calls are handled, which depends on the phone’s RIL (Radio Interface Layer) and telephony libraries. On GSIs, especially Unisoc devices, the stock vendor files are tightly tied to the modem, so just copying random lib*.so files usually won’t fix it.

For incoming calls, the important files are things like:

libvendor-ril.so or libril.so – handles call registration and signaling

libims.so – needed if your carrier uses VoLTE

Files like libwpa_client.so or librpmbclient.so aren’t related to calls—they handle Wi-Fi or SMS.

The easiest ways to fix it:

Use a GSI that explicitly supports Unisoc T606 and includes the proper vendor blobs.

If you want to patch manually, you need the exact RIL and IMS files from your stock ROM—just grabbing random libraries won’t work.

Make sure your modem/baseband firmware matches your ROM, or the phone won’t register incoming calls.

In short: the problem isn’t trivial, and fixing it requires the correct vendor libraries from your stock firmware, not guessing which files to copy.

**Reply by soncao0212-commits** [reply]
@usmanamin3254 friend, you seem very professional, please help me, I have a headache about this problem. my processor is unisoc t157, android 13 and I have the original rom. it can't receive calls when someone calls, functions like messages work fine. I don't know what files to copy, what files are RIL and IMS and when I have it, how to patch it, rebuild the rom or what. please help me, I will be very grateful. don't let me continue to fumble in the dark. please.

**Reply by soncao0212-commits** [reply]
@usmanamin3254 please reply me soon please god bless you
